.github/actions/setup/action.yml

```diff
@@ -0,0 +1,26 @@
+name: 'Setup'
+description: 'Install Node, PNPM, and dependencies using cache'
+runs:
+    using: composite
+    steps:
+        - uses: actions/setup-node@2fddd8803e2f5c9604345a0b591c3020ee971a93 # tag=v3
+        - uses: pnpm/action-setup@35ab4267a1a21c8e8cb1c087cf1642e891ff57bd
+          with:
+              version: 7
+              run_install: false
+        - name: Get pnpm store directory
+          id: pnpm-cache
+          shell: bash
+          run: |
+              echo "::set-output name=pnpm_cache_dir::$(pnpm store path)"
+        - uses: actions/cache@v3
```

Comment:
we can because plasma is open source? 😁  Noice!

Reply:
Yup, we can also use the github hosted runners so it costs us nothing to run

Reply:
<img src="https://media3.giphy.com/media/hAcDHEhZHA2bu/giphy.gif"/> that's cool 😎